heft-plugins/heft-serverless-stack-plugin/src/ServerlessStackPlugin.ts

```diff
@@ -5,142 +5,152 @@ import * as path from 'path';
+              this._logger.terminal.writeVerboseLine(
+                'The command line includes "--sst", redirecting Webpack to Serverless Stack'
+              );
+              return null;
+            }
+          );
+        }
+      );
 
-    const sstStageParameter: IHeftStringParameter = heftSession.commandLine.registerStringParameter({
-      associatedActionNames: ['test', 'build', 'start'],
-      parameterLongName: '--sst-stage',
-      argumentName: 'STAGE_NAME',
-      description:
-        'Specifies the Serverless Stack stage; equivalent to to the "--stage" parameter from the "sst" CLI'
-    });
+      taskSession.requestAccessToPluginByName(
+        '@rushstack/heft-webpack5-plugin',
+        Webpack5PluginName,
+        async (accessor: IWebpack5PluginAccessor) => {
+          accessor.onConfigureWebpackHook?.tapPromise(
+            { name: PLUGIN_NAME, stage: Number.MAX_SAFE_INTEGER },
+            async (config: IWebpack5Configuration | null) => {
+              // Discard Webpack's configuration to prevent Webpack from running
+              this._logger.terminal.writeVerboseLine(
+                'The command line includes "--sst", redirecting Webpack to Serverless Stack'
+              );
+              return null;
+            }
+          );
+        }
+      );
 
-    let sstCliPackagePath: string;
+      taskSession.hooks.run.tapPromise(PLUGIN_NAME, async (runOptions: IHeftTaskRunHookOptions) => {
+        // TODO: Handle watch / serve mode
+        await this._runServerlessStackAsync({
+          taskSession,
+          heftConfiguration,
+          sstStage: sstStageParameter.value,
+          debugMode: taskSession.debugMode,
+          // serveMode:
+          verbose: runOptions.verbose
+        });
+      });
+    }
+  }
 
+  private async _runServerlessStackAsync(options: {
+    taskSession: IHeftTaskSession;
+    heftConfiguration: HeftConfiguration;
+    sstStage?: string;
+    debugMode?: boolean;
+    serveMode?: boolean;
+    verbose?: boolean;
+  }): Promise<void> {
+    let sstCliPackagePath: string;
     try {
       sstCliPackagePath = Import.resolvePackage({
         packageName: SST_CLI_PACKAGE_NAME,
-        baseFolderPath: heftConfiguration.buildFolder
+        baseFolderPath: options.heftConfiguration.buildFolder
       });
     } catch (e) {
       throw new Error(
-        `The ${TASK_NAME} task cannot start because your project does not seem to have a dependency on the` +
-          ` "${SST_CLI_PACKAGE_NAME}" package: ` +
+        `The ${options.taskSession.taskName} task cannot start because your project does not seem to have ` +
+          `a dependency on the "${SST_CLI_PACKAGE_NAME}" package: ` +
           e.message
       );
     }
 
     const sstCliEntryPoint: string = path.join(sstCliPackagePath, 'bin/scripts.js');
-    if (!FileSystem.exists(sstCliEntryPoint)) {
+    if (!(await FileSystem.existsAsync(sstCliEntryPoint))) {
```

Comment:
Can you stick this in a variable instead of having an `await` call inside an `if` block?